packages/react-router/src/routeInfo.ts

```diff
@@ -53,22 +55,34 @@ export type CurrentPath<TOption> = 'always' extends TOption
     ? '.'
     : './' | '.'
 
-export type CatchAllPaths<TOption> =
-  | CurrentPath<TOption>
-  | ParentPath<TOption>
-  | ''
+export type CatchAllPaths<TOption> = CurrentPath<TOption> | ParentPath<TOption>
 
-export type RoutesByPath<TRouteTree extends AnyRoute> = {
-  [K in ParseRoute<TRouteTree> as K['fullPath']]: K
-}
+export type CodeRoutesByPath<TRouteTree extends AnyRoute> =
+  ParseRoute<TRouteTree> extends infer TRoutes extends AnyRoute
```

Comment:
You'll see this trick used a lot. It basically makes `ParseRoutes` lazily instantiated so that when we go down the file based routing path, the route tree is not parsed at all